src/Symfony/Component/Security/Http/HttpUtils.php

```diff
@@ -135,6 +135,10 @@ public function generateUri($request, $path)
             return $path;
         }
 
+        if (!$request instanceof Request) {
```

Comment:
Problem is it also lacks a object typehint (`Request $request`). We should prefer that over docblocks of course.. however the current policy doesnt seem to allow it.. (?)

https://github.com/symfony/symfony/pull/19960#discussion_r81571700 closed for the same reason. Allthough i still disagree ;-)